File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/model/tsp/service/information/ServiceSupplyPoint.java
package tr.gov.tubitak.uekae.esya.api.tsl.model.tsp.service.information;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

import tr.gov.tubitak.uekae.esya.api.tsl.TSLException;
import tr.gov.tubitak.uekae.esya.api.tsl.model.core.BaseElement;
import tr.gov.tubitak.uekae.esya.api.tsl.util.Constants;
import tr.gov.tubitak.uekae.esya.api.tsl.util.XmlUtil;

public class ServiceSupplyPoint extends BaseElement {
	private String mServiceSupplyPoint;

	public ServiceSupplyPoint(Document document, String iSSPoint)

	{
		super(document);
		mServiceSupplyPoint = iSSPoint;
		mElement.setTextContent(mServiceSupplyPoint);
	}

	public ServiceSupplyPoint(Document document)

	{
		super(document);
	}

	public ServiceSupplyPoint(Element aElement) throws TSLException

	{
		super(aElement);
		mServiceSupplyPoint = XmlUtil.getText(aElement);
	}

	@Override
	public String getLocalName() {
		return Constants.TAG_SERVICESUPPLYPOINT;
	}

	@Override
	public String getNamespace() {

		return Constants.NS_TSL;
	}

	public String getSSPoint() {
		return mServiceSupplyPoint;
	}

	public void setSSPoint(String value) {
		mServiceSupplyPoint = value;
		mElement.setTextContent(mServiceSupplyPoint);
	}
}

File: api-tsl/csharp/src/model/tsp/service/information/ServiceSupplyPoint.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using tr.gov.tubitak.uekae.esya.api.common.util;
using tr.gov.tubitak.uekae.esya.api.tsl.model.core;
using tr.gov.tubitak.uekae.esya.api.tsl.util;

namespace tr.gov.tubitak.uekae.esya.api.tsl.model.tsp.service.information
{
    public class ServiceSupplyPoint : BaseElement
    {
        private string mServiceSupplyPoint;

        public ServiceSupplyPoint(XmlDocument document, string iSSPoint)
            : base(document)
        {
            mServiceSupplyPoint = iSSPoint;
            mElement.InnerText = mServiceSupplyPoint;
        }

        public ServiceSupplyPoint(XmlDocument document)
            : base(document)
        {
        }

        public ServiceSupplyPoint(XmlElement aElement)
            : base(aElement)
        {
            mServiceSupplyPoint = XmlUtil.getText(aElement);
        }

        public override string LocalName
        {
            get { return Constants.TAG_SERVICESUPPLYPOINT; }
        }

        public override string Namespace
        {
            get
            {
                return Constants.NS_TSL;
            }
        }

        public string SSPoint
        {
            get { return mServiceSupplyPoint; }
            set
            {
                mServiceSupplyPoint = value;
                mElement.InnerText = mServiceSupplyPoint;
            }
        }
    }
}
